autorest/azure/async.go

```diff
@@ -37,6 +38,80 @@ const (
 	operationSucceeded  string = "Succeeded"
 )
 
+// Future provides a mechanism to access the status and results of an asynchronous request.
+// Since futures are stateful they should be passed by value to avoid race conditions.
```

Comment:
This language still make me a little uneasy. I'd be more comfortable adding a Mutex of some sort. However, if you still feel strongly, I will defer.

Reply:
If we do that then it makes everybody pay that tax (albeit a small one).  Keeping it as-is leaves it up to the consumer whether or not they need this functionality which is why documenting that it's stateful is important.

Reply:
I can reword this though and am open to suggestions.